drivers/SmartThings/matter-rvc/src/init.lua

```diff
@@ -93,6 +88,20 @@ local function do_configure(driver, device)
 
   device.log.info_with({hub_logs = true}, string.format("Updating device profile to %s.", profile_name))
   device:try_update_metadata({profile = profile_name})
+end
+
+local function device_init(driver, device)
+  device:subscribe()
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  if not device:get_field(SERVICE_AREA_PROFILED) and #device:get_endpoints(clusters.ServiceArea.ID) > 0 then
+    match_profile(driver, device)
+    device:set_field(SERVICE_AREA_PROFILED, true)
+  end
+end
+
+local function do_configure(driver, device)
+  match_profile(driver, device)
+  device:set_field(SERVICE_AREA_PROFILED, true)
```

Comment:
```suggestion
  device:set_field(SERVICE_AREA_PROFILED, true, { persist = true })
```
this field should not be reset on a driver restart